pkg/overlay/service.go

```diff
@@ -139,7 +147,7 @@ func (s *Service) Process(ctx context.Context, _ *cobra.Command, _ []string) (
 		return err
 	}
 
-	kad, err := kademlia.NewKademlia(id, []proto.Node{*in}, "0.0.0.0", localPort)
+	kad, err := kademlia.NewKademlia(id, []proto.Node{*in}, "0.0.0.0", viper.GetString("localport"))
```

Comment:
every config value inside pkg/overlay should probably start with "overlay." perhaps? like "overlay.localport"?